deep_groove_project/src/main.py

```diff
@@ -1,53 +1,106 @@
 import torch
-import altair as alt
-import matplotlib.pyplot as plt
 import pandas as pd
-
+from torch import nn
+from torch.utils.data import DataLoader
+from src.FeatureDataset import FeatureDataset
+from src.NeuralNetwork import NeuralNetwork
+import matplotlib.pyplot as plt
 
 def main():
-    df = pd.read_csv("../data/first_dataset.csv")
-    show_standard_info(df)
-    show_correlation_grafs(df)
-    show_3d_scatter(df)
-
-
-def show_standard_info(df):
-    print("\n General Info:")
-    df.info()
-    print("\n Table:")
-    print(df.head())
-
-
-def show_correlation_grafs(df):
-    fields = ['Fr','n','Lifetime']
-
-    chart = alt.Chart(df).mark_point().encode(
-        alt.X(alt.repeat("column"), type='quantitative'),
-        alt.Y(alt.repeat("row"), type='quantitative')
-    ).properties(
-        width=200,
-        height=200
-    ).repeat(
-        row=fields,
-        column=fields[::1]
-    ).interactive()
-    chart.show()
-
-
-def show_3d_scatter(df):
-    fig = plt.figure()
-    ax = fig.add_subplot(projection='3d')
-
-    fr = df.iloc[:, 0]
-    n = df.iloc[:, 1]
-    lifetime = df.iloc[:, 2]
-
-    ax.scatter(fr, n, lifetime)
-    ax.set_xlabel('Fr')
-    ax.set_ylabel('n')
-    ax.set_zlabel('Lifetime')
+    device = "cuda" if torch.cuda.is_available() else "cpu"
+    model = NeuralNetwork().to(device)
+    print(model)
+
+    #  Loading and preparing Dataframe
+    dataframe = pd.read_csv("../data/first_dataset.csv")
+
+    # shuffle Dataframe
+    dataframe = dataframe.sample(frac=1)
+
+    print(dataframe.head())
+    dataframe['Label'] = 0
+    dataframe['Label'].where(dataframe['Lifetime'] <= 8760, 1, inplace=True)
+    dataframe['Label'].where(dataframe['Lifetime'] <= 8760*2, 2, inplace=True)
+    dataframe['Label'].where(dataframe['Lifetime'] <= 8760*3, 3, inplace=True)
+    dataframe['Label'].where(dataframe['Lifetime'] <= 8760*4, 4, inplace=True)
+    dataframe['Label'].where(dataframe['Lifetime'] <= 8760*4, 5, inplace=True)
+
+    print(dataframe.head())
+    # Creating custom Dataset
+    train_dataset = FeatureDataset(dataframe[:int(len(dataframe)/2)])
+    test_dataset = FeatureDataset(dataframe[int(len(dataframe)/2):])
+
+    print(f"length of Datasets - training: {len(train_dataset)}, test: {len(test_dataset)}")
+
+    # Parameters
+    params = {'batch_size': 10,
+              'shuffle': True}
+
+    # Creating the Dataloaders
+    train_dataloader = DataLoader(train_dataset, **params)
+    test_dataloader = DataLoader(test_dataset, **params)
+
+    for batch in train_dataloader:
+        print(f"Batch: {batch[0]}")
+        break
+
+    learning_rate = 1e-7
+    print(f"LR:  {learning_rate}")
+
+    loss_fn = nn.CrossEntropyLoss()
```

Comment:
It is great to see, that you structured your code in a pure python file. The next step is to move all files belongig to the first_dataset into its own subfolder.

Feel free to find a good naming convention for the name of the subfolder like "classification_dataset1"

Reply:
For now it is okay!